src/script/lua_api/l_object.cpp

```diff
@@ -72,8 +72,6 @@ PlayerSAO* ObjectRef::getplayersao(ObjectRef *ref)
 		return NULL;
 	if (obj->getType() != ACTIVEOBJECT_TYPE_PLAYER)
 		return NULL;
-	if (obj->isGone())
-		return NULL;
 	return (PlayerSAO*)obj;
```

Comment:
How about Peer ID checks here?

Reply:
Peer ID invalidation always happens together with marking the object as "gone":
https://github.com/minetest/minetest/blob/4e2473ec493689c3bc7d619fe17ee31c7f20945b/src/server/player_sao.cpp#L555-L559